### ISSUE TITLE
Adjusted height & depth of elements moved via voffset

### ISSUE THREAD

**Issue by ronkok** [w3c/mathml-core#203]
The `<mpadded>` element has a `voffset` attribute, which enables one to shift the vertical alignment of an element. However, the MathML-Core specification makes no mention of what happens to the `height` and `depth` of an element that is so shifted.

In practice, neither Chromium nor Gecko currently changes the `height` or `depth` of an element that is shifted via `voffset`. In MathML 3.0, one could manually adjust the `height` and `depth` via attribute adjustments such as `height="+6pt"`. MathML core does not have such a `±` adjustment.

Some way is needed to make that adjustment of `height` and `depth`. Otherwise, one will see elements impinging on fraction bars, radical bars, and the like. 

Currently in Chromium, one can apply CSS padding and that will increase the `height` or the `depth` of the element, depending on whether the padding is applied to the top or the bottom of the element.  Gecko does not behave this way. So I know of no way to currently write an adjustment to `height` and `depth` that will run correctly in both browsers.

Is the padding hack a valid way to address this dilemma? Will it continue to run in future versions of Chromium and perhaps the other browsers? Should this hack be explicitly written into the MathML Core specification?

Or will some other means of `height` adjustment become part of MathML-Core? Perhaps a return of the `±` notation?

**Comment by ronkok** [comment]
Or should the adjustment of `height` and `depth` happen automatically whenever `voffset` is applied? This would also overcome some problems I have observed in which shifted math overruns its container.

**Comment by NSoiffer** [comment]
There are a number of problems caused by the implementations not being in sync. I've run into the CSS issue for trying to write a [polyfill for `menclose`](https://w3c.github.io/mathml-polyfills/menclose/https://w3c.github.io/mathml-polyfills/menclose/). The polyfill makes use of CSS to draw arrowheads and works nicely in Chrome. In FireFox, no CSS so examples 16 - 28 are basically no-ops. The results would be even worse in Firefox but since it implements `menclose`, I do browser sniffing and skip running the polyfill in the cases where Firefox has an implementation.

I recall @dginev running into this also, but I don't remember the example.

Missing CSS implementations notwithstanding, I think it would be a reasonable default to shift height and depth automatically because that is what (I think) people normally want.

**Comment by bkardell** [comment]
Minutes from April 27 2026
```
NS issue 203 hight and depth cannot be changed in core....Can it just be closed?
Action @eerii check to see if item 203 can be closed because hight and depth are supported in all the browsers, ideally provide a wpt link showing we have tests and it can be closed.

```

**Comment by ronkok** [comment]
Yes, `height` and `depth` are currently supported. That is insufficient to resolve this issue. The need is for a way to increment `height` or `depth`. Remember, the MathML 2.0 increment syntax is not supported by MathML Core. An expression such as `depth=+0.5em` does not increment the height by `0.5em`.

**Comment by dginev** [comment]
Cross-referencing that #306 nominated using CSS `calc-size()` for relative adjustments, once it stabilizes. Likely something on the lines:

```css
depth: calc-size(auto, size + 0.5em);
```

There was also [an interop issue](https://github.com/web-platform-tests/interop/issues/1008) for that which wasn't approved, maybe next year.

**Comment by NSoiffer** [comment]
I am doing some work on the mfraca@bevelled polyfill and ran into a problem with the spec/implementation constraints that the height and depth cannot be negative. This is a little different from Ron's issue in that the polyfill can compute the height and depth (although much more work than it should be). 

In the following example, I'm trying to shift the denominator up24px (a large amount (for debugging). The polyfill wants to keep the bounding box around the numerator, so it adds to the height and subtracts from the depth (not computed correctly in this example, but close enough). But this makes the depth negative.

In the image below, the green bounding box is around the mpadded and as can be seen, is anything but tight around the box because the depth was clipped to zero. I don't think there is a browser limitation that requires the spec/implementation to clip to 0 for height and depth (in the debugger, I can adjust the bounding box to be tight). The limitation should be that `height + depth >=0`, not individually.

<img width="969" height="395" alt="Image" src="https://github.com/user-attachments/assets/762d25c8-4f2e-4403-bb70-c8155be03359" />

@fred-wang, @eerii: is it possible to change the implementation? If so, then I think the spec should change.
